src/raiden_libs/blockchain.py

```diff
@@ -95,9 +94,9 @@ def query_blockchain_events(
     Args:
         web3: A Web3 instance
         contract_manager: A contract manager
-        contract_address: The address of the contract to be filtered, can be `None`
+        contract_addresses: The address of the contract to be filtered, can be a single address
```

Comment:
This single-address-or-list is a bit strange. We could always take a list and call it with single element lists where necessary. If we keep both options, I would probably prefer the singular for the parameter name. No need to change this if you prefer it this way, though.

Reply:
Yeah, I was unhappy with that as well, I chose the union because the single address is still common, but on the second thought it seems like the better idea to only have a list as the parameter.